resource-aggregate/cqrs/eventstore/mongodb/eventstore.go

```diff
@@ -635,183 +755,79 @@ func (s *EventStore) Close(ctx context.Context) error {
-		opts.SetHint(eventsQueryIndex)
-		return bson.D{{Key: aggregateIDKey, Value: aggregateID2Hash("snapshot")}, {Key: versionKey, Value: -1}}, &opts
-	}
-
-	if len(queries) == 1 {
-		opts := options.FindOptions{}
-		opts.SetHint(snapshotsQueryIndex)
-		return bson.D{{Key: idKey, Value: getSnapshotID(queries[0].AggregateID)}}, &opts
-	}
-
-	orQueries := make([]bson.M, 0, 32)
-	for _, q := range queries {
-		if q.AggregateID != "" {
-			orQueries = append(orQueries, bson.M{idKey: getSnapshotID(q.AggregateID)})
+func makeDBEvents(events []eventstore.Event, marshaler MarshalerFunc) ([]bson.M, error) {
+	dbEvents := make([]bson.M, 0, len(events))
+	for idx, event := range events {
+		// Marshal event data if there is any.
+		raw, err := marshaler(event)
+		if err != nil {
+			return nil, fmt.Errorf("cannot create db event from event[%v]: %w", idx, err)
 		}
+		dbEvents = append(dbEvents, bson.M{
+			versionKey:    event.Version(),
+			dataKey:       raw,
+			eventTypeKey:  event.EventType(),
+			isSnapshotKey: event.IsSnapshot(),
+		})
 	}
-	opts := options.FindOptions{}
-	opts.SetHint(snapshotsQueryIndex)
-	return bson.M{"$or": orQueries}, &opts
+	return dbEvents, nil
 }
 
-type queryIterator struct {
-	iter    *mongo.Cursor
-	groupID string
-}
-
-func (i *queryIterator) Next(ctx context.Context, q *eventstore.VersionQuery) bool {
-	var query bson.M
-
-	if !i.iter.Next(ctx) {
-		return false
-	}
-
-	err := i.iter.Decode(&query)
-	if err != nil {
-		return false
-	}
-
-	version := query[dataKey].(int64)
-	q.Version = uint64(version)
-	q.AggregateID = query[aggregateIDStrKey].(string)
-	q.GroupID = i.groupID
-	return true
-}
-
-func (i *queryIterator) Err() error {
-	return i.iter.Err()
-}
-
-func (s *EventStore) loadSnapshotQueries(ctx context.Context, groupID string, queries []eventstore.SnapshotQuery, qh *loader) error {
-	var err error
-	var iter *mongo.Cursor
-	query, hint := snapshotQueriesToMgoQuery(queries)
-	if hint == nil {
-		iter, err = s.client.Database(s.DBName()).Collection(getEventCollectionName(groupID)).Find(ctx, query)
-	} else {
-		iter, err = s.client.Database(s.DBName()).Collection(getEventCollectionName(groupID)).Find(ctx, query, hint)
-	}
-	if err == mongo.ErrNilDocument {
-		return nil
-	}
-	if err != nil {
-		return err
-	}
-	if s.goroutinePoolGo != nil {
-		err = qh.QueryHandlePool(ctx, &queryIterator{iter: iter, groupID: groupID})
-	} else {
-		err = qh.QueryHandle(ctx, &queryIterator{iter: iter, groupID: groupID})
-	}
-	errClose := iter.Close(ctx)
-	if err == nil {
-		return errClose
-	}
-	return nil
-}
-
-// RemoveUpToVersion deletes the aggragates events up to a specific version.
-func (s *EventStore) RemoveUpToVersion(ctx context.Context, queries []eventstore.VersionQuery) error {
-	collections := make(map[string][]eventstore.VersionQuery)
-	for _, query := range queries {
-		collections[query.GroupID] = append(collections[query.GroupID], query)
+// RemoveUpToVersion deletes the aggregates events up to a specific version.
```

Comment:
Can we split this file into multiple ones? 800+ lines ...

Reply:
fixed